[VDCA-584] Added Org system tests

rocknes opened the pull request:
* Added Org system tests.
* Added proper license header to all the system test files.
* Standardized Client.get_org_by_name() and Client.get_org_list() method's output format
* Handled fallout of the changes to Client.get_org_by_name() and Client.get_org_list()'s output.
* Fixed a bug in Utils.get_admin_href() and Utils.get_admin_extension_href() to make them idempotent.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vmware/pyvcloud/258)
<!-- Reviewable:end -->

hodgesrm commented:
Reviewed 9 of 9 files at r1.
Review status: all files reviewed, 4 unresolved discussions (waiting on @rocknes)

---

*[pyvcloud/vcd/client.py, line 1137 at r1](https://reviewable.io/reviews/vmware/pyvcloud/258#-LFu6ukP89YBhZ32x5f7:-LFu6ukQ3NdeADGu_d79:b-p28hp2) ([raw file](https://github.com/vmware/pyvcloud/blob/9b85bfa966ac4e42d2045616fba9c0261df6f817/pyvcloud/vcd/client.py#L1137)):*
> ```Python
>         """
>         # Avoid using get_org_list() to fetch all orgs and then filter the
>         # reuslt by organization name, since get_org_list() will fetch details
> ```

reuslt -> result

---

*[pyvcloud/vcd/utils.py, line 607 at r1](https://reviewable.io/reviews/vmware/pyvcloud/258#-LFu7IWL352h0HJ8gXXO:-LFu7IWL352h0HJ8gXXP:blin6s5) ([raw file](https://github.com/vmware/pyvcloud/blob/9b85bfa966ac4e42d2045616fba9c0261df6f817/pyvcloud/vcd/utils.py#L607)):*
> ```Python
> 
> def get_admin_href(href):
>     """Returns admin version of a given vCD url.
> ```

Should note that if handed an admin URL you will return it without change (to match case logic below).

---

*[system_tests/org_tests.py, line 101 at r1](https://reviewable.io/reviews/vmware/pyvcloud/258#-LFu7nL20h3YB3_waa8U:-LFu7nL20h3YB3_waa8V:bss4nnd) ([raw file](https://github.com/vmware/pyvcloud/blob/9b85bfa966ac4e42d2045616fba9c0261df6f817/system_tests/org_tests.py#L101)):*
> ```Python
>         except EntityNotFoundException as e:
>             return
>         self.fail('Should fail with EntityNotFoundException while fetching'
> ```

This code looks unreachable.  You should have an exception other than EntityNotFoundException which means Python will unwind the stack before hitting this line. 

---

*[system_tests/org_tests.py, line 125 at r1](https://reviewable.io/reviews/vmware/pyvcloud/258#-LFu8MYl1ndjQDHpLPMQ:-LFu8MYl1ndjQDHpLPMR:b-yld6oe) ([raw file](https://github.com/vmware/pyvcloud/blob/9b85bfa966ac4e42d2045616fba9c0261df6f817/system_tests/org_tests.py#L125)):*
> ```Python
>         """Test the  method System.delete_org().
> 
>         Invoke delete operation on an enabled organization.
> ```

Nit: Had to read this twice to be clear that the operation would fail because the Org is enabled and force=False.  Should clarify in comment. 

---


*Comments from [Reviewable](https://reviewable.io/reviews/vmware/pyvcloud/258)*
<!-- Sent from Reviewable.io -->

rocknes commented:
Review status: 6 of 9 files reviewed, 4 unresolved discussions (waiting on @hodgesrm and @rocknes)

---

*[pyvcloud/vcd/client.py, line 1137 at r1](https://reviewable.io/reviews/vmware/pyvcloud/258#-LFu6ukP89YBhZ32x5f7:-LFx9Zj_COx_JJ_IB_W7:b-886gsx) ([raw file](https://github.com/vmware/pyvcloud/blob/9b85bfa966ac4e42d2045616fba9c0261df6f817/pyvcloud/vcd/client.py#L1137)):*
<details><summary><i>Previously, hodgesrm (Robert Hodges) wrote…</i></summary><blockquote>

reuslt -> result
</blockquote></details>

Fixed

---

*[pyvcloud/vcd/utils.py, line 607 at r1](https://reviewable.io/reviews/vmware/pyvcloud/258#-LFu7IWL352h0HJ8gXXO:-LFxA3KY26jNpZ_m95JU:beknk1m) ([raw file](https://github.com/vmware/pyvcloud/blob/9b85bfa966ac4e42d2045616fba9c0261df6f817/pyvcloud/vcd/utils.py#L607)):*
<details><summary><i>Previously, hodgesrm (Robert Hodges) wrote…</i></summary><blockquote>

Should note that if handed an admin URL you will return it without change (to match case logic below).
</blockquote></details>

Added extra doc sting explaining the behavior

---

*[system_tests/org_tests.py, line 125 at r1](https://reviewable.io/reviews/vmware/pyvcloud/258#-LFu8MYl1ndjQDHpLPMQ:-LFxCdvl0aOuYj84zGZc:b-j5eqg4) ([raw file](https://github.com/vmware/pyvcloud/blob/9b85bfa966ac4e42d2045616fba9c0261df6f817/system_tests/org_tests.py#L125)):*
<details><summary><i>Previously, hodgesrm (Robert Hodges) wrote…</i></summary><blockquote>

Nit: Had to read this twice to be clear that the operation would fail because the Org is enabled and force=False.  Should clarify in comment. 
</blockquote></details>

Added additional doc-string to clarify the test case.

---


*Comments from [Reviewable](https://reviewable.io/reviews/vmware/pyvcloud/258)*
<!-- Sent from Reviewable.io -->

hodgesrm commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>  Thanks for the changes.  Looks great. 

---

Reviewed 3 of 3 files at r2.
Review status: :shipit: complete! all files reviewed, all discussions resolved

---



*Comments from [Reviewable](https://reviewable.io/reviews/vmware/pyvcloud/258#-:-LFyJ3jkE8mGWQlbphmi:bld0w9o)*
<!-- Sent from Reviewable.io -->
